### PR TITLE
Update GaudiNIC multi-node-training dockerfile and setup

### PR DESCRIPTION
**Pull request by yeonsily**
# What does this PR do?

Fixes # (issue)

## Before submitting
- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [ ] Did you make sure to update the documentation with your changes?
- [ ] Did you write any new necessary tests?

### CODE REVIEW

**Review by libinta — COMMENTED** [review]
since we give example of 4 nodes, should we update the hostfile template with 4 nodes?

File: examples/multi-node-training/README.md [hunk 1]
```diff
@@ -44,8 +46,26 @@ docker run -it --runtime=habana -e HABANA_VISIBLE_DEVICES=all -e OMPI_MCA_btl_va
 
 > For AWS DL1 instances, `--privileged` must be passed to the `docker run` command so that EFA interfaces are visible.
 
-Finally, you will have to copy the public key of the leader node in the `~/.ssh/authorized_keys` file of all other nodes to enable password-less SSH.
 
+You will have to copy id_rsa.pub contents from every node's docker to every other node's docker's ~/.ssh/authorized_keys (all public keys need to be in all hosts' authorized_keys) to enable password-less SSH.
```
Comment (regisss) [inline examples/multi-node-training/README.md:50]: Weird I never had to do that on AWS DL1 instances with EFA

**Review by regisss — COMMENTED** [review]
I think we should also have a `multi-node-training/efa/` folder where we store the other Dockerfile and `.deepspeed_env` and clearly state in the README that we provide images for both Gaudi NIC and EFA cases.

File: examples/multi-node-training/README.md (hunk 1 above)
Comment (yeonsily) [reply examples/multi-node-training/README.md:53]: Created EFA folder and move AWS dockerfile and deepspeed_env to there. 
But this step is mentioned previously also and I just explained step by step details.

Comment (regisss) [reply examples/multi-node-training/README.md:53]: Yes but in my understanding we only need to copy the key of the leader node to every other bode's authorized keys, not every key.

**Review by regisss — COMMENTED** [review]
We also need to update this link: https://github.com/huggingface/optimum-habana/blob/0fb6bb6b7bbfb06b85a6c10f113b8283aa028b02/docs/source/usage_guides/multi_node_training.mdx#L126

File: examples/multi-node-training/README.md (hunk 1 above)
Comment (yeonsily) [reply examples/multi-node-training/README.md:53]: I had to follow the step from https://github.com/HabanaAI/Model-References/tree/master/PyTorch/nlp/DeepSpeedExamples/deepspeed-bert#set-up-password-less-ssh for GaudiNIC.

Let me try just leader node key also.

Comment (yeonsily) [reply examples/multi-node-training/README.md]: I confirmed that we only need to copy leader node's key. I updated the README.

**Review by regisss — APPROVED** [review]

**Pull request merged**